builtin/logical/aws/secret_access_keys.go

```diff
@@ -89,6 +90,39 @@ func (b *backend) secretAccessKeysCreate(
 			"Error creating access keys: %s", err)), nil
 	}
 
+	// IAM is eventually consistent, so give it time to settle; if it doesn't
+	// within a reasonable time, error out. 20 should be plenty.
+	loops := 10
+	for {
+		loops--
+		if loops == 0 {
+			return logical.ErrorResponse(fmt.Sprintf(
+				"IAM failed to report a valid user")), nil
+		}
+		guo, err := client.GetUser(&iam.GetUserInput{
+			UserName: aws.String(username),
+		})
+		if err != nil {
+			awsErr, ok := err.(awserr.Error)
+			if !ok {
+				return logical.ErrorResponse(fmt.Sprintf(
+					"unexpected error type returned from AWS SDK: %T, err is %v", err, err),
+				), nil
+			}
+			if awsErr.Code() != "NoSuchEntity" {
+				return logical.ErrorResponse(fmt.Sprintf(
+					"unexpected error returned from AWS: %v", err),
+				), nil
+			}
+		} else {
+			if guo.User == nil {
+				return logical.ErrorResponse(fmt.Sprintf(
+					"user returned from AWS was nil")), nil
+			}
```

Comment:
You're totally right. I also realized, after seeing this, that I did in fact run the acceptance tests on this backend, but forgot that it had not been merged in for 0.4, so they didn't include this change (this can be merged in for 0.5). I'll fix that now and update.